### ISSUE TITLE
Feature request: iwyu_tool.py should exit with a non-zero exit value if it has detected include issues

### ISSUE THREAD

**Issue by Flowdalic** [include-what-you-use/include-what-you-use#790]
As of now I have to check `iwyu_tool.py`'s output to determine if there are any issues with the include. Which not only requires extra work, it is also fragile, as the output the tool emits in case there are include issues may change.

It would be great, if  `iwyu_tool.py` would simply exit with a non-zero exit value if it has detected include issues.

**Comment by kimgr** [comment]
I think it boils back down to the confusion in `include-what-you-use` itself -- by design it always returns a failure exit code, so that `make -k` can be used to re-run it.

I'm not convinced that decision was a good one, so maybe `iwyu-tool.py` should at least try and report success/failure consistently.

I think we should do something about `include-what-you-use` itself too, but I'm not sure what, especially not to maintain backward compat. Maybe a `--fail-always` switch to produce the current behavior.

**Comment by Flowdalic** [comment]
> Maybe a `--fail-always` switch to produce the current behavior.

If you are concerned about backwards compatibility, which is always good, then a `--fail-only-on-iwyu-error` switch is probably a better idea.

**Comment by kimgr** [comment]
Yeah, but I think the current behavior is weird enough that it warrants explicit user action to preserve :)

**Comment by mortie** [comment]
I just decided to check out IWYU, and this bit me too. I would love to require `make iwyu` to return 0 in order to commit, and check that sort of stuff in CI, but that's really hard due to the current design.

I use GNU make, so the `make -k` hack is unnecessary to me since `-B` exists which seems to do the job better. It would probably be best to just require `--fail-always` for the special case of BSD make users who want to run `make CC=iwyu` ad-hoc from the command line, since it makes the tool so much more useful for everyone (including BSD make users).
